src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -0,0 +1,435 @@
+            The activation used by the layer.
+        reduction (`int`, *optional*, defaults to 4):
+            The reduction factor the block applies in the first `1x1` convolution.
+    """
+
+    def __init__(
+        self, in_channels: int, out_channels: int, stride: int = 1, activation: str = "relu", reduction: int = 4
+    ):
+        super().__init__()
+        should_apply_shortcut = in_channels != out_channels or stride != 1
+        reduces_channels = out_channels // reduction
+        self.shortcut = (
+            ResNetShortCut(in_channels, out_channels, stride=stride) if should_apply_shortcut else nn.Identity()
+        )
+        self.layer = nn.Sequential(
+            ResNetConvLayer(in_channels, reduces_channels, kernel_size=1),
+            ResNetConvLayer(reduces_channels, reduces_channels, stride=stride),
+            ResNetConvLayer(reduces_channels, out_channels, kernel_size=1, activation=None),
+        )
+        self.activation = ACT2FN[activation]
+
+    def forward(self, pixel_values):
+        residual = pixel_values
+        pixel_values = self.layer(pixel_values)
+        residual = self.shortcut(residual)
+        pixel_values += residual
+        pixel_values = self.activation(pixel_values)
+        return pixel_values
+
+
+class ResNetStage(nn.Sequential):
+    """
+    A ResNet stage composed by stacked layers.
+
+    Args:
+        in_channels (`int`):
+            Number of input channels.
+        out_channels (`int`):
+            Number of output channels.
+        stride (`int`, *optional*, defaults to 1):
+            The first layer stride, used to downsample the input.
+        depth (`int`,*optional*, defaults to 2):
+            The number of layers.
+        layer_type (`str`, *optional*, defaults to `"basic"`):
+            The layer to use, it can be either `"basic"` (used for smaller models, like resnet-18 or resnet-34) or
+            `"bottleneck"` (used for larger models like resnet-50 and above).
+        activation (`int`, *optional*, defaults to `"relu"`):
+            The activation used by all layers.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        stride: int = 2,
+        depth: int = 2,
+        layer_type: str = "basic",
+        activation: str = "relu",
+    ):
+        super().__init__()
+
+        layer = ResNetBottleNeckLayer if layer_type == "bottleneck" else ResNetBasicLayer
+
+        self.layers = nn.Sequential(
+            # downsampling is done in the first layer with stride of 2
+            layer(in_channels, out_channels, stride=stride, activation=activation),
+            *[layer(out_channels, out_channels, activation=activation) for _ in range(depth - 1)],
+        )
+
+
+class ResNetEncoder(nn.Module):
+    def __init__(self, config: ResNetConfig):
+        super().__init__()
+        self.stages = nn.ModuleList([])
+        # the first block doesn't downsample
```

Comment:
(nit) do you mean the first stage? Or just the first block of the first stage

Reply:
The first block of the first stage (in our case first layer). To be update